src/TK-Quake.Engine/Infrastructure/Abstract/ILivableEntity.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TKQuake.Engine.Infrastructure.Math;
+
+namespace TKQuake.Engine.Infrastructure.Abstract
+{
+    /// <summary>
+    /// A livable entity that resides in the game world such as the player or NPC
+    /// </summary>
+    public interface ILivableEntity
+    {
+        void Move(Vector position);
+
+        /// <summary>
+        /// Spanws the player in the world with initial health
```

Comment:
Nitpick: Some of these comments are specific to a player when they should be more general.